### ISSUE TITLE
ANN: Deprecating project! May the best fork win!

### ISSUE THREAD

**Issue by taoeffect** [tjfontaine/node-dns#111]
Per issue #96, the deadline for @tjfontaine to respond has well since passed, and since:

- I haven't heard anything from him
- I do not have the time to maintain this project myself
- And I cannot add new owners to this project (don't have the privs)

I'm declaring this project dead and inviting the community to get behind an actually-maintained fork.

I'm updating the README with a link to this thread, and if the community can get itself to rally behind a replacement I will post a link in the README to the new project.

Now... SHOW ME WHAT YOU GOT!

[![screen shot 2016-02-13 at 1 31 29 pm](https://cloud.githubusercontent.com/assets/138706/13030191/372586da-d256-11e5-8b72-612b930ebd25.jpg)](https://www.youtube.com/watch?v=FzQbJAKVmt0)

**Comment by fastest963** [comment]
I'll throw out that I'm willing to maintain this project (under my company Leven Labs) to the best of my ability. I'm maintaining just over 20 packages on npm (https://www.npmjs.com/~fastest963) of which some of them utilize dns and I'll be moving over to use node-dns.

To be honest though, I've just started using this module so if someone else with more direct experience steps up, they have my vote.

**Comment by taoeffect** [comment]
Ah, a fellow Gator, cool. :)

Thanks for commenting! I'll keep this issue open for at least one more week to give others a chance to respond.

**Comment by taoeffect** [comment]
@fastest963 OK, looks like you are the winner by default! Congratulations! :smile: 

Have you forked this repo yet? I recommend renaming it so that there's no conflict in NPM. Please let me know what I should link the README to.

**Comment by taoeffect** [comment]
Seems @fastest963 dropped out, so the search for a maintainer continues!

**Comment by roccomuso** [comment]
any news?

**Comment by julianxhokaxhiu** [comment]
It seems that this guy did a fork 21 days ago and proactively changed everything to tell people that is now maintained by him: https://github.com/tutosfaciles48/node-dns

Although I need to issue a bug report I'm having with `dig -6` and `bad packet` but I don't know where to submit it, since the current maintainer do not longer provide support. Too bad :(

**Comment by taoeffect** [comment]
If @tutosfaciles48 is serious about maintaining this project please let me know.

**Comment by coolaj86** [comment]
@tjfontaine 30 seconds of your time, PLEASE:

1. Go to https://github.com/tjfontaine/node-dns/settings
2. Click `Transfer` in the Danger Zone
3. Move to **organization** `Daplie` as `node-dns` (that way we can have teams with `owners`)

```
npm owner add coolaj86 native-dns
```

I don't care which organization it's moved to or who becomes co-owner, we just need you to pick someone to transfer it to so that we have a way to move forward.

If you transfer to me and add me to npm my plan is to add people that are actively working on bugfixes so that they can contribute and to bump the major version for safety (since this has been ~~stagnant~~ stable for a very long time).

**Comment by danger89** [comment]
Good luck! I hope the project won't die eventually. I love your work. Kudo's to all of you!

**Comment by fralec** [comment]
Any news?

**Comment by AaronNGray** [comment]
@fastest963 - any news ?

**Comment by brodybits** [comment]
@coolaj86 according to https://docs.npmjs.com/misc/disputes and https://www.npmjs.com/policies/disputes you should be able to get co-ownership of the `node-dns` or `native-dns` package **if** you follow the procedure they described. In short (from https://www.npmjs.com/policies/disputes):
> 1. Get the author email with npm owner ls <pkgname>
> 1. Email the author, CC support@npmjs.com
> 1. After a few weeks, if there's no resolution, we'll sort it out.
> Don't squat on package names. Publish code or move out of the way.

**Comment by taoeffect** [comment]
Although @coolaj86 could do what @brodybits suggested, he could also do as mentioned in the very post in this issue: find someone willing to maintain the project. Then I will update the README to point to that project, which will have a new name.

**Comment by taoeffect** [comment]
It doesn't make sense to transfer the project to someone who is not already willing to maintain it (and has demonstrated this by adding code / merging pull requests / etc.). You'll just end up in the same place as we are now.

**Comment by coolaj86** [comment]
@taoeffect agreed

**Comment by OmgImAlexis** [comment]
I've sent in an email to @tjfontaine and hopefully I get a reply soon, if not I'm just going to keep trying to get in contact with him until this is sorted as I'd like to use this on a project I'm working on but I'm hesitant while it's not being maintained by anyone. 😞

**Comment by mcollina** [comment]
@OmgImAlexis at this point you can contact NPM support and ask for the module name.
There is a process for that, and it works well. They will try to contact him themselves.

**Comment by OmgImAlexis** [comment]
@mcollina that only helps with getting the npm module name sorted, still doesn't fix the issue of people forking from this repo. Even with a new repo established as the "correct" one people will still end up here and forks will still be made from this. It'd be best to have this moved to an org and have someone set as an admin so they can take the project to where it should be.

**Comment by mcollina** [comment]
@OmgImAlexis once the naming of npm is sorted, the field over there will get updated. As far as I know, @taoeffect has the commit bit here, so he can wipe this repo clean.
People will reach from NPM to the new github repo easily.

**Comment by taoeffect** [comment]
Yeah, @mcollina is right, and at risk of sounding like a broken record: show us your fork that demonstrates you're serious about maintaining this project, and I'll update the README to point it in your direction.

**Comment by mcollina** [comment]
I think you can move things forward here: ask for the npm package, and then
you can crown someone. At this point anyone it's better than nobody.
Il giorno mer 5 apr 2017 alle 19:48 Greg Slepak <notifications@github.com>
ha scritto:

**Comment by taoeffect** [comment]
> At this point anyone it's better than nobody.

As a security person, I can tell you with certainty that is not true. :P

**Comment by cyberbizness** [comment]
https://github.com/tjfontaine/node-dns/issues/117

**Comment by WoZ** [comment]
I may propose to use our module that was tested in production and highload environments. Also we have good unit and functional tests [with 100% coverage](https://coveralls.io/github/LCMApps/dns-lookup-cache?branch=master), multi-records resolving and TTL support. Here it is: https://github.com/LCMApps/dns-lookup-cache

**Comment by FrancisTurner** [comment]
I'm considering stepping up and maintaining this. Or at least looking at the existing pull requests and merging them into my fork. If they don't break anything then I'll probably add a couple of other things I want too (APL RRtype for one)

**Comment by FrancisTurner** [comment]
I have merged three pull requests into my fork. If there is demand I can merge others. Feel free to submit stuff to it.

**Comment by taoeffect** [comment]
Thank you @FrancisTurner. I've linked to your comment in the README. 😄 

https://github.com/tjfontaine/node-dns#dead-see-forks

**Comment by FrancisTurner** [comment]
Maybe add a link to the github repo too ? https://github.com/FrancisTurner/native-node-dns

**Comment by taoeffect** [comment]
Sure. Added.

**Comment by eduardoruizm** [comment]
A new fork with DNSSEC:
https://github.com/eduardoruizm/native-dnssec-dns

**Comment by EduardoRuizM** [comment]
A new fork with DNSSEC and IANA parameters:
https://github.com/EduardoRuizM/native-dnssec-dns

**Comment by taoeffect** [comment]
Congrats @titanism and thank you for letting us know! I've added a link to the README pointing to Tangerine 😄

**Comment by taoeffect** [comment]
@EduardoRuizM I've also added a comment to your fork as well - sorry for not noticing earlier!

**Comment by EduardoRuizM** [comment]
Thanks @taoeffect
